infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -84,7 +86,15 @@ public class CorfuReplicationDiscoveryService implements Runnable, CorfuReplicat
      * It is backed by a corfu store table.
      **/
     @Getter
-    private LogReplicationMetadataManager logReplicationMetadataManager;
+    private final Map<String, LogReplicationMetadataManager>
+        logReplicationMetadataManagerMap = new HashMap<>();
```

Comment:
Another thing, though this maybe for future : I think we should check if it's possible to make the sinkManager and MetadataManager classes even smaller since the number of objects is going to scale with the number of remote sites.